openapi_tester/schema_tester.py

```diff
@@ -185,20 +185,24 @@ def _validate_key_casing(
             tester(key)
 
     @staticmethod
-    def _validate_enum(schema_section: dict, data: str) -> Union[Optional[str], bool]:
-        if "enum" not in schema_section:
+    def _validate_enum(schema_section: dict, data: Any) -> Union[Optional[str], bool]:
+        enum = schema_section.get("enum")
+
+        if enum is None:
             return False
 
-        enum = schema_section.get("enum")
-        if enum and data not in enum:
+        if data not in enum:
             return (
                 f'Mismatched values, expected a member of the enum {schema_section["enum"]} but received {str(data)}.'
             )
 
     @staticmethod
-    def _validate_pattern(schema_section: dict, data: str) -> Union[Optional[str], bool]:
-        if "pattern" not in schema_section:
+    def _validate_pattern(schema_section: dict, data: Any) -> Union[Optional[str], bool]:
```

Comment:
Why return a union type from validators? This shoukd be consisteny - optional[str]

Reply:
Agree - done in e111565